.github/workflows/ci-e2e-cassandra.yml

```diff
@@ -48,9 +48,39 @@ jobs:
       run: bash scripts/cassandra-integration-test.sh ${{ matrix.version.major }} ${{ matrix.version.schema }} ${{ matrix.jaeger-version }}
       env:
         SKIP_APPLY_SCHEMA: ${{ matrix.create-schema == 'auto' && true || false }}
+     
+    - name: Upload Metrics Artifact
```

Comment:
please move to a shared action under .github/actions/

Reply:
a shared action for the whole thing from upload artifact to cache metric or just the upload artifact ?

Reply:
a shared action that either uploads artifact or adds to a cache. The action can decide those two choices based on github.ref, it does not need help from the actual workflow. I would expect the **only** change to the e2e workflows to look like that
```
uses: ./.github/actions/save-metrics-snapshot
if: ${{ matrix.jaeger-version }} == "v2"
with:
  snapshot: ./.metrics/cassandra_metrics.txt
  artifact_key: ...
  cache_key: ...
```